### ISSUE TITLE
Unity to Neos conversion

### ISSUE THREAD

**Issue by Frooxius** [Frooxius/NeosPublic#52]
Polish up the internal Unity to Neos converter and provide as part of an SDK, to allow exporting Unity scenes and objects into Neos easily.

**Comment by riftopia** [comment]
Wow, this is an amazing feature, it brings a whole world of possibilities to NeosVR!

**Comment by Casuallynoted** [comment]
Down the road, avatar import support may be neat in the SDK as well- unless that would be too much work to add, but perhaps a means to translate other avatar scripts to Neos avatar components.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Frooxius** [comment]
Yes avatar conversion would be possible. The way I want to make is so you can write your own conversion functions for components. That way community could create and share modules to convert lots of different things that aren't included with the SDK.

**Comment by Enverex** [comment]
The SDK works well right now, but a little back to front. It's fine for importing a few objects/models, but not a project because the way it currently works is that you add your items to this SDK's own project, wherein reality you'd expect to be able to add this SDK to your existing project to be able to import it.

**Comment by Frooxius** [comment]
Currently the SDK is not even proper "SDK", it's just expanded version of tool I wrote for myself in very early Neos alpha stages to transfer some objects to Neos during the development, but that I made available to Patreon supporters.

The plan for the proper SDK is to essentially fully integrate it with Unity and make it much easier to operate. So you just import it to your project, right click part of hierarchy or the world and click "Convert to Neos" and it pretty much does the rest for you.

The conversion code itself will more or less be similar, but I would also like to make it extensible, so you can write custom conversion routines for things it doesn't recognize by default, so community can add support for more things as well as you can add support for custom parts of your project.

I would also expand what it supports by default to particle systems, audio, animations, terrains (once we have terrain system) and other common things.

**Comment by dfgHiatus** [comment]
Though not directly related, I have a tool that facilities the import of Unity Packages: https://github.com/dfgHiatus/NeosUnityPackagesImporter
